src/main/java/org/wso2/lsp4intellij/utils/GUIUtils.java

```diff
@@ -80,12 +85,40 @@ public static Hint createAndShowEditorHint(Editor editor, String string, Point p
     public static Hint createAndShowEditorHint(Editor editor, String string, Point point, short constraint, int flags) {
         JTextPane textPane = new JTextPane();
         textPane.setEditorKit(new HTMLEditorKit());
+        // add spaces to the start and end of code blocks as padding
+        string = string.replace("<code>", "<code>&nbsp;")
+                .replaceAll("(?<!\\n)</code>", "&nbsp;</code>");
+
         textPane.setText(string);
+        HTMLDocument doc = (HTMLDocument) textPane.getDocument();
+        StyleSheet styleSheet = doc.getStyleSheet();
+
+        styleSheet.addRule("p { font-family: " + "Segoe UI Semibold" + "; font-size: 10px; }");
+        styleSheet.addRule("code { font-family: " + "Arial" + "; font-weight: bold; font-size: 10px;}");
+        for (int i = 1; i <= 6; i++) {
+            styleSheet.addRule("h" + i + " { font-family: " + "Arial" + "; font-weight: bold;}");
+        }
+
+        // add theme aware background color to code blocks
+        Color bodyFontColor = styleSheet.getStyle("body").getAttribute(StyleConstants.Foreground) instanceof Color ?
+                (Color) styleSheet.getStyle("body").getAttribute(StyleConstants.Foreground) : Color.BLACK;
+
+        Color inverseColor = new Color(255 - bodyFontColor.getRed(),
+                255 - bodyFontColor.getGreen(),
+                255 - bodyFontColor.getBlue()).darker();
+
+        String hexColor =
+                String.format("#%02x%02x%02x", inverseColor.getRed(), inverseColor.getGreen(), inverseColor.getBlue());
+        styleSheet.addRule("code { background-color: " + hexColor + "; }");
+
```

Comment:
I prefer to refactor the `createAndShowEditorHint()` method into a set of utils, instead doing everything inside this single function?